src/AbstractSQLRules2SQL.ts

```diff
@@ -889,6 +910,19 @@ const typeRules: Dictionary<MatchFn> = {
 		const replacement = TextValue(getAbstractSqlQuery(args, 2), indent);
 		return `REPLACE(${str}, ${find}, ${replacement})`;
 	},
+	ExtractJSONPathAsText: (args, indent) => {
+		checkMinArgs('ExtractJSONPathAsText', args, 1);
+		if (engine !== Engines.postgres) {
+			throw new SyntaxError(
+				'ExtractJSONPathAsText not supported on: ' + engine,
+			);
+		}
+		const column = TextValue(getAbstractSqlQuery(args, 0), indent);
```

Comment:
This should be a `JSONValue` (`isJSONValue` will need updating to include `ToJSON` and be wrapped in a `MatchValue`), trying JSON operators on a text type won't work, also renamed as `json` rather than `column` since it doesn't need to be a column and can just be any JSON type
```suggestion
		const json = JSONValue(getAbstractSqlQuery(args, 0), indent);
```